app-misc/piper/piper-0.4.ebuild

```diff
@@ -0,0 +1,44 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit meson python-single-r1 udev
+
+DESCRIPTION="GTK configuration application for libratbag"
+HOMEPAGE="https://github.com/libratbag/piper"
+SRC_URI="https://github.com/libratbag/piper/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+BDEPEND="
+	virtual/pkgconfig
+"
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-libs/libratbag
+	$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
+	$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]')
```

Comment:
Additional dependencies are needed:
```
$ grep -Er '(gi\.require_version|from gi\.repository import)' *
piper/application.py:gi.require_version("Gio", "2.0")
piper/application.py:gi.require_version("Gtk", "3.0")
piper/application.py:from gi.repository import Gio, GLib, Gtk  # noqa
piper/buttondialog.py:gi.require_version("Gtk", "3.0")
piper/buttondialog.py:from gi.repository import Gdk, GObject, Gtk  # noqa
piper/buttonspage.py:gi.require_version("Gtk", "3.0")
piper/buttonspage.py:from gi.repository import Gtk  # noqa
piper/devicerow.py:gi.require_version("Gtk", "3.0")
piper/devicerow.py:from gi.repository import GdkPixbuf, GObject, Gtk, Rsvg  # noqa
piper/errorperspective.py:gi.require_version("Gtk", "3.0")
piper/errorperspective.py:from gi.repository import GObject, Gtk  # noqa
piper/gi_composites.py:from gi.repository import Gio, GLib, GObject, Gtk  # noqa
piper/leddialog.py:gi.require_version("Gtk", "3.0")
piper/leddialog.py:from gi.repository import Gdk, GObject, Gtk  # noqa
piper/ledspage.py:gi.require_version("Gtk", "3.0")
piper/ledspage.py:from gi.repository import Gtk  # noqa
piper/mousemap.py:gi.require_version("Gdk", "3.0")
piper/mousemap.py:gi.require_version("Gtk", "3.0")
piper/mousemap.py:gi.require_version("Rsvg", "2.0")
piper/mousemap.py:from gi.repository import Gdk, GLib, Gtk, GObject, Rsvg  # noqa
piper/mouseperspective.py:gi.require_version("Gtk", "3.0")
piper/mouseperspective.py:from gi.repository import GLib, GObject, Gtk  # noqa
piper/optionbutton.py:gi.require_version("Gtk", "3.0")
piper/optionbutton.py:from gi.repository import Gtk  # noqa
piper/profilerow.py:gi.require_version("Gtk", "3.0")
piper/profilerow.py:from gi.repository import GObject, Gtk  # noqa
piper/ratbagd.py:from gi.repository import Gio, GLib, GObject  # noqa
piper/resolutionrow.py:gi.require_version("Gtk", "3.0")
piper/resolutionrow.py:from gi.repository import GObject, Gtk  # noqa
piper/resolutionspage.py:gi.require_version("Gtk", "3.0")
piper/resolutionspage.py:from gi.repository import Gtk  # noqa
piper/svg.py:from gi.repository import Gio  # noqa
piper/welcomeperspective.py:gi.require_version("Gtk", "3.0")
piper/welcomeperspective.py:from gi.repository import GObject, Gtk, Gdk  # noqa
piper/window.py:gi.require_version("Gtk", "3.0")
piper/window.py:from gi.repository import Gdk, GLib, Gtk, Gio  # noqa
piper.in:gi.require_version('Gio', '2.0')
piper.in:gi.require_version('Gtk', '3.0')
piper.in:from gi.repository import Gio, Gtk
```

Many packages install GObject-Introspection files only conditionally with USE="introspection", so `[introspection]` will be needed for them.
In this case:
- `gnome-base/librsvg[introspection]` is needed for `/usr/share/gir-1.0/Rsvg-2.0.gir`.
- `x11-libs/gdk-pixbuf[introspection]` is needed for `/usr/share/gir-1.0/GdkPixbuf-2.0.gir`.
- `x11-libs/gtk+:3[introspection]` is needed for `/usr/share/gir-1.0/Gdk-3.0.gir` and `/usr/share/gir-1.0/Gtk-3.0.gir`.
- You may add `dev-libs/gobject-introspection` for `Gio`, `GLib` and `GObject`, although it is guaranteed to be installed by other dependencies...

Above dependencies are needed at least in `RDEPEND`.

Reply:
https://github.com/libratbag/piper/blob/0ab5c63ed404f619dcd6c380d8881d3f9eb3d406/meson.build#L73 and https://github.com/libratbag/piper/blob/65c740ced2960548b286b7beaf49a2a798f36be3/piper/mousemap.py#L20 say that `dev-python/lxml` is needed.

https://github.com/libratbag/piper/blob/0ab5c63ed404f619dcd6c380d8881d3f9eb3d406/meson.build#L75 and https://github.com/libratbag/piper/blob/65c740ced2960548b286b7beaf49a2a798f36be3/piper/mousemap.py#L17 say that `dev-python/pycairo` is needed.

Reply:
Glad you pointed out [introspection], I missed then when trying to find these missing depends on my own.  Fixed.